exonum/benches/block.rs

```diff
@@ -106,13 +107,14 @@ mod tests {
     fn execute_cryptocurrency(db: Box<Database>, b: &mut Bencher) {
         let mut blockchain = create_blockchain(db);
 
-        message! {
-            struct Tx {
-                const TYPE = 1;
-                const ID = 0;
+        transactions! {
```

Comment:
And what about more "rusty" syntax? I think that this is quite possible to do.
```rust
transactions! {
    enum Transactions {
        const SERVICE_ID: u16 = 0;

        TxFirst {
            from: &PublicKey,
            data: &[u8]
        },
        TxSecond {
            from: &PublicKey,
            time: SystemTime,
        }
    }
}
```

Reply:
It was one option I was considering, but I decided against it in the end:

The really important fact is that each of the transactions is a type on its own, it is really a standalone struct. 

The fact that there's a wrapping enum is not really important: you never work with it as with an enum, you only ever use it in the boiler plate implementation of `Service::tx_from_raw` and when parsing transaction from JSON, but both of this places should be handled automatically once we parametrize Service over `TransactionSet` (as in #424).

On the other hand, it's quite nice that you `impl Transaction` on a case by case basis.